src/MessageHandler/MbinMessageHandler.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\MessageHandler;
+
+use App\Message\Contracts\MessageInterface;
+use Doctrine\DBAL\Exception;
+use Doctrine\ORM\EntityManagerInterface;
+
+abstract class MbinMessageHandler
+{
+    public function __construct(
+        private readonly EntityManagerInterface $entityManager,
+    ) {
+    }
+
+    /**
+     * @throws \Throwable
+     * @throws Exception
+     */
+    public function workWrapper(MessageInterface $message): void
+    {
+        $conn = $this->entityManager->getConnection();
+        if (!$conn->isConnected()) {
+            $conn->connect();
+        }
+
+        $conn->transactional(fn () => $this->doWork($message));
```

Comment:
Ah it was a bit hidden away in 58 files changed, but this is the wrapper method that is now calling `transactional`, which allows you to take over the transaction demarcation.

Reply:
This also means that no flush needs to be called anymore in the `doWork`.